include/phoenix/vm.hh

```diff
@@ -530,43 +532,49 @@ namespace phoenix {
 				} else {
 					throw vm_exception {"unsupported return type"};
 				}
-			} else {
+			} else if (!std::is_same_v<naked_call, R>) {
 				if (sym->has_return())
 					throw illegal_external_rtype(sym, "void");
 			}
 
 			std::vector<symbol*> params = find_parameters_for_function(sym);
-			if (params.size() < sizeof...(P))
+			if (params.size() < sizeof...(P) && !std::is_same_v<naked_call, R>)
 				throw illegal_external_definition {sym,
 				                                   "too many arguments declared for function override " + sym->name() +
 				                                       ": declared " + std::to_string(sizeof...(P)) + " expected " +
 				                                       std::to_string(params.size())};
 
-			if (params.size() > sizeof...(P))
+			if (params.size() > sizeof...(P) && !std::is_same_v<naked_call, R>)
 				throw illegal_external_definition {sym,
 				                                   "not enough arguments declared for function override " +
 				                                       sym->name() + ": declared " + std::to_string(sizeof...(P)) +
 				                                       " expected " + std::to_string(params.size())};
 
-			if constexpr (sizeof...(P) > 0) {
+			if constexpr (sizeof...(P) > 0 && !std::is_same_v<naked_call, R>) {
 				check_external_params<0, P...>(params);
 			}
```

Comment:
Do I understand correctly that you never want to take any parameters or return any values to the VM when using a naked call? In that case, this I feel like this should be a separate function since it uses basically none of the logic of `override_function`.

Reply:
Typically naked-calls are special case, so explicit parameter pooling is good match for feature; yet must admit that for loop it's not mandatory

Reply:
Okay so then let's split that logic out into a `override_function_naked` routine like this:

```cpp

void override_function_naked(std::string_view name, std::function<void(vm&)> callback) {
    auto* sym = find_symbol_by_name(name);
    if (sym == nullptr)
        throw vm_exception {"symbol not found"};
    if (sym->is_external())
        throw vm_exception {"symbol is already an external"};

    _m_function_overrides[sym->address()] = [callback, sym](vm& machine) {
        callback(machine);
    };
}
```

That makes this considerably easier to read I think.

Reply:
I've set for solution in-between;
add new override for `override_function(..., std::function<naked_call(vm&))` - so code can be decoupled.
decide not to introduce new name, as on user end `phoenix::naked_call` is already a strong hint.